### PR TITLE
test: add e2e test cases for route with metric plugin

### PR DESCRIPTION
**Pull request by nic-chen**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [x] E2E test

- Related issues

#636

### CODE REVIEW

**Review by membphis — COMMENTED** [review]
function and file name should contain "Prometheus"

File: api/test/e2e/route_with_metric_plugin_test.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+		{
+			caseDesc:     "request from client (200)",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "hello world",
+		},
+		{
+			caseDesc: "create route that uri not not exists in upstream",
```
Comment (gxthrj) [inline api/test/e2e/route_with_metric_plugin_test.go:74]: double `not`

File: api/test/e2e/route_with_metric_plugin_test.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+			}`,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+		},
+		{
+			caseDesc:     "request from client (404)",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Path:         "/hello-not-exists",
+			ExpectStatus: http.StatusNotFound,
```
Comment (gxthrj) [inline api/test/e2e/route_with_metric_plugin_test.go:100]: Does the upstream return 404 when uri is /hello-not-exists ?

Comment (nic-chen) [reply api/test/e2e/route_with_metric_plugin_test.go:100]: yes. it is.

File: api/test/e2e/route_with_metric_plugin_test.go (hunk 1 above)
Comment (nic-chen) [reply api/test/e2e/route_with_metric_plugin_test.go]: fixed.

**Comment by nic-chen** [comment]
> function and file name should contain "Prometheus"

OK, the function indeed should  contain "Prometheus",  and updated.file name I think could be metric , other test cases for metric plugin could write into it too.

**Review by membphis — APPROVED** [review]
LGTM

**Comment by nic-chen** [comment]
ping @gxthrj

**Review by gxthrj — APPROVED** [review]

**Pull request merged**